### ISSUE TITLE
Yio remote custom client

### ISSUE THREAD

**Issue by tex-cyb** [SEPIA-Framework/sepia-docs#135]
If you're willing to support it I'd love to see support for the yio remote , either with a custom client or with a rest api trigger designed for it , that way I can use sepia with the remote as opposed to something like alexa or Google assistant , and if the always on feature with avatar support is possible on the remote I'd love to see it as well

**Comment by fquirin** [comment]
Sounds like a great idea :grinning: ... unfortunately I don't have a Yio remote for testing :-/.
Nevertheless there should be multiple options to control SEPIA with the remote. Some things that come to my mind:
- Send HTTP request from Yio to SEPIA via [remote-actions endpoint](https://github.com/SEPIA-Framework/sepia-docs/blob/master/API/assist-server.md#remote-actions)
- Use the (work-in-progress) [Node-RED interface for SEPIA](https://github.com/SEPIA-Framework/sepia-node-js-client/tree/dev) somehow
- Use a [custom service or widget](https://github.com/SEPIA-Framework/sepia-extensions) somehow
- Use the [smart-home HUB features](https://github.com/SEPIA-Framework/sepia-docs/wiki/Smart-Home-Controls) somehow, but the remote is probably not a device to be controlled. Might still be interesting to read data from the remote.

> ...and if the always on feature with avatar support is possible on the remote I'd love to see it as well

This would be awesome :star_struck: ... but I think the remote doesn't have enough juice to run the full client :thinking: . I'm currently preparing some experiments with the new Raspberry Pi Zero 2 to test if it works well enough (it should but 512MB RAM is probably the absolute minimum possible). For less powerful systems I think one has to write a new SEPIA client that focusses on super low resource consumption. The [Node.js code](https://github.com/SEPIA-Framework/sepia-node-js-client/tree/dev) I started for Node-RED could be a starting point.

> I'll post an update when they respond

Yes please :-)

**Comment by tex-cyb** [comment]
The remote will be getting support for voice assistants , i just haven't gotten a response as to how exactly they implement it into the remote, my best guess is they use a skill for said smart assistant

**Comment by tex-cyb** [comment]
YIO Remote Two, via @Kickstarter https://www.kickstarter.com/projects/marton/remote-two?ref=android_project_share , here's the link , and also I'd be more than happy to lend you mine once I get it , i bought it as the portable control hub for a future project that i can't carry out currently

**Comment by tex-cyb** [comment]
Here are a few other interesting pieces of hardware I've backed that might mesh well with sepia ,Axon | WiFi + LoRa Messaging Platform made for IoT, via @Kickstarter https://www.kickstarter.com/projects/amin-mehdi-mansouri/axon-wifi-messaging-platform-made-for-iot?ref=android_project_share ,UDOO KEY: The Flexible AI Platform, From $4, via @Kickstarter https://www.kickstarter.com/projects/udoo/udoo-key-the-4-ai-platform?ref=android_project_share, I'd be more than willing to lend them to you if you'd like

**Comment by tex-cyb** [comment]
> Sounds like a great idea 😀 ... unfortunately I don't have a Yio remote for testing :-/. Nevertheless there should be multiple options to control SEPIA with the remote. Some things that come to my mind:
> 
> * Send HTTP request from Yio to SEPIA via [remote-actions endpoint](https://github.com/SEPIA-Framework/sepia-docs/blob/master/API/assist-server.md#remote-actions)
> * Use the (work-in-progress) [Node-RED interface for SEPIA](https://github.com/SEPIA-Framework/sepia-node-js-client/tree/dev) somehow
> * Use a [custom service or widget](https://github.com/SEPIA-Framework/sepia-extensions) somehow
> * Use the [smart-home HUB features](https://github.com/SEPIA-Framework/sepia-docs/wiki/Smart-Home-Controls) somehow, but the remote is probably not a device to be controlled. Might still be interesting to read data from the remote.
> 
> > ...and if the always on feature with avatar support is possible on the remote I'd love to see it as well
> 
> This would be awesome 🤩 ... but I think the remote doesn't have enough juice to run the full client 🤔 . I'm currently preparing some experiments with the new Raspberry Pi Zero 2 to test if it works well enough (it should but 512MB RAM is probably the absolute minimum possible). For less powerful systems I think one has to write a new SEPIA client that focusses on super low resource consumption. The [Node.js code](https://github.com/SEPIA-Framework/sepia-node-js-client/tree/dev) I started for Node-RED could be a starting point.
> 
> > I'll post an update when they respond
> 
> Yes please :-)

It might be possible to create a custom client that runs on more powerful hardware and the remote just uses a interface to communicate with the sepia hub client

**Comment by fquirin** [comment]
> It might be possible to create a custom client that runs on more powerful hardware and the remote just uses a interface to communicate with the sepia hub client

Yes, that is certainly an option :-). So far I've focused on the HTML client because I simply had no capacity left to support other versions, but I'm planning to build a super-low resource client for a long time now. In general the hardest technical challenge is the audio streaming which could maybe be done via Linux's Pulseaudio (it actually supports remote devices) or something like gStreamer etc.. The rest would "simply" be extensive event handling and UI ^^.
